src/05-derivatives/02-common-data-types.md

```diff
@@ -1,4 +1,115 @@
+[The Inheritance Principle](../02-common-principles.md#the-inheritance-principle))
+unless a particular derivative includes REQUIRED metadata fields, in which case a
+JSON file is also REQUIRED.
+Each derivative type defines their own set of fields, but all of them
+share the following (non-required) ones:
+
+| **Key name** | **Description**                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
+| ------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Description  | RECOMMENDED. Free-form natural language description of the nature of the file.                                                                                                                                                                                                                                                                                                                                                                                                                                             |
+| Sources      | OPTIONAL. A list of paths relative to dataset root pointing to the file(s) that were directly used in the creation of this derivative. For example, if a derivative A is used in the creation of another derivative B, which is in turn used to generate C in a chain of A->B->C, C should only list B in `Sources`, and B should only list A in `Sources`. However, in case both X and Y are directly used in the creation of Z, then Z should list X and Y in `Sources`, regardless of whether X was used to generate Y. |
+| RawSources   | OPTIONAL. A list of paths relative to dataset root pointing to the BIDS-Raw file(s) that were used in the creation of this derivative. When the derivative filename does not define a `space` keyword, the first entry of `RawSources` MUST be defined and it will define the `scanner` coordinate system that applies.                                                                                                                                                                                                    |
```

Comment:
"... the first entry of `RawSources` MUST be defined ..."  I suggest changing this "defined" to "specified".

Reply:
Part of my confusion remains, due to the `OPTIONAL` which then later turns into `MUST`. Here is an attempt to rewrite:

> REQUIRED or OPTIONAL: If the derived file does not contain `space`, it is required and the first file in the list defines the coordinate system; if the derived file does contain `space`, it is optional.

Another concern is that it is unclear to me how RawSources would be used with a derived EEG data file (which are also covered by this common specification), or a derived behav-only data file (also part of the BIDS specification). The raw source EEG data file that it would point to does not define a `scanner` coordinate system. I believe that this description can only be applied to derived data files (and corresponding source files) that define some spatial coordinate system. As such I do see how this can apply to MEG (e.g. following Maxfilter).

The validator should not flag derived EEG or behavioural datasets without `RawSources` and without `space` keyword as invalid.

Reply:
Talking with @oesteban I think this use of `RawSources` to define a reference has been made obsolete by `SpatialReference`. I will remove this language, and adjust the language in the example.